Automattic/wp-calypso
# Add domain-only upsell carousel in All domains page

## Changes proposed in this Pull Request

This PR adds an upsell carousel in `All domains` management page. 
It's part of the new domain-only flow (pcYYhz-ye-p2)

This carousel shows two separate cards - create site and add email - for the most recent domain-only purchase.

The email card is showed only if there is no email services (Titan or Gmail) associated to the domain.

_Note 1: illustrations are placeholder, I'm going to replace them before merging (they are not ready yet)._

_Note 2: the dismiss card feature will be added later in a separate PR._

![carousel site](https://user-images.githubusercontent.com/2797601/153248192-ac3a68d5-e142-4a03-8be4-88027cd6baf0.png)

![carousel email](https://user-images.githubusercontent.com/2797601/153248178-3078d657-d30f-4d81-a7c3-5c059d9d58d2.png)

## Testing instructions

_Prerequisites: you need a domain-only subscriptions to test this PR_

- Build this branch locally or open the live Calypso link
- Navigate to to `/manage/domains`
- Verify that the carousel is showed and CTAs are working correctly
- Verify that the email card is not showed if the domain has an associated email subs (you can simulate add a subscription through React devtools)
